Sources/Drill/DiagnosticRegexParser.swift

```diff
@@ -0,0 +1,94 @@
+/// DiagnosticRegexParser.swift
+///
+/// Copyright 2017, The Silt Language Project.
+///
+/// This project is released under the MIT license, a copy of which is
+/// available in the repository.
+
+import Foundation
+
+/// A RegexPiece is either:
+///  - A literal string that must be matched exactly.
+///  - A regular expression that must be matched using regex matching.
+/// To construct a regular expression from these pieces, `literal` pieces
+/// must be escaped such that they will always match special regex
+/// characters literally.
+enum RegexPiece {
+  /// The string inside must be matched by the resulting regex literally.
+  case literal(String)
+
+  /// The string inside is a regex that must be added wholesale into the
+  /// resulting regex.
+  case regex(String)
+
+  /// Regex-escapes the piece appropriately, taking into account the need
+  /// to escape special characters in literals.
+  var asRegex: String {
+    switch self {
+    case .literal(let str):
+      let requiresEscape = Set("*?+[(){}^$|\\./")
+      var s = ""
+      for c in str {
+        if requiresEscape.contains(c) {
+          s.append("\\")
+        }
+        s.append(c)
```

Comment:
NSRegularExpression can do this for you.

Reply:
Oh thank god, I looked for a method for this but couldn't find anything.

Reply:
Ah oops, found it.